include/circt/Dialect/Ibis/IbisOps.td

```diff
@@ -160,13 +167,48 @@ def CallOp : IbisOp<"call", [CallOpInterface]> {
   }];
 }
 
+def GetParentOp : IbisOp<"get_parent"> {
+  let summary = "Ibis get parent";
+  let description = [{
+    Given an Ibis class reference, returns the parent of said class. The
+    returned `ibis.classref` will be an opaque type.
+  }];
+
+  let arguments = (ins AnyClassRefType:$src);
```

Comment:
To support getting the grand parent of an instance, the argument must accept either type.